projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/user/entity/User.kt

```diff
@@ -218,20 +223,194 @@ data class Caseload(
     val team: Team,
 
     @Column(name = "role_code")
-    val roleCode: String
+    val roleCode: String,
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumns(
+        JoinColumn(name = "offender_id", referencedColumnName = "offender_id", insertable = false, updatable = false),
+        JoinColumn(name = "staff_employee_id", referencedColumnName = "staff_id", insertable = false, updatable = false)
+    )
+    val nextAppointment: NextAppointment? = null,
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumns(
+        JoinColumn(name = "offender_id", referencedColumnName = "offender_id", insertable = false, updatable = false),
+        JoinColumn(name = "staff_employee_id", referencedColumnName = "staff_id", insertable = false, updatable = false)
+    )
+    val previousAppointment: PreviousAppointment? = null,
+
+    @ManyToOne
+    @JoinColumn(name = "offender_id", insertable = false, updatable = false)
+    val latestSentence: LatestSentence? = null
+
+)
+
+@Entity
+@Subselect(
+    """
+        select * from(
+            select e.*,
+            row_number() over (partition by e.offender_id order by cast(e.event_number as number) desc) as row_num 
+            from event e join main_offence mo on mo.event_id = e.event_id
+            where e.soft_deleted = 0 
+            and e.active_flag = 1
+            ) sub
+        where sub.row_num = 1
+"""
+)
+@Synchronize("event")
+data class LatestSentence(
+
+    @Id
+    @Column(name = "offender_id")
+    val id: Long,
+
+    @ManyToOne
+    @JoinColumn(name = "event_id", referencedColumnName = "event_id", insertable = false, updatable = false)
+    val mainOffence: MainOffence? = null,
 )
 
+@Entity
+@Subselect(
+    """
+        select * from (
+            select 
+            q.staff_id,
+            q.offender_id,
+            q.appointment_datetime,
+            q.contact_id,
+            q.contact_type_id,
+            row_number() over (partition by staff_id, offender_id order by appointment_datetime asc) as row_num 
```

Comment:
Why do we need to partition by staff id here?  Are we only showing the latest/next appointment for the current user with the person on probation?

Reply:
we are actually looking for the latest appointment for the staff record here as we are looking for the staffs appointment with the crn. The appointment for the crn could in theory be with another staff member